src/main/java/wooteco/chess/domain/piece/position/Direction.java

```diff
@@ -0,0 +1,105 @@
+	E(0, 1, (rowGap, columnGap) -> rowGap == 0 && columnGap > 0),
+	W(0, -1, (rowGap, columnGap) -> rowGap == 0 && columnGap < 0),
+
+	NE(1, 1, (rowGap, columnGap) -> isSameAbs(rowGap, columnGap) && rowGap > 0 && columnGap > 0),
+	NW(1, -1, (rowGap, columnGap) -> isSameAbs(rowGap, columnGap) && rowGap > 0 && columnGap < 0),
+	SE(-1, 1, (rowGap, columnGap) -> isSameAbs(rowGap, columnGap) && rowGap < 0 && columnGap > 0),
+	SW(-1, -1, (rowGap, columnGap) -> isSameAbs(rowGap, columnGap) && rowGap < 0 && columnGap < 0),
+
+	NNE(2, 1, (rowGap, columnGap) -> rowGap == 2 && columnGap == 1),
+	NNW(2, -1, (rowGap, columnGap) -> rowGap == 2 && columnGap == -1),
+	SSE(-2, 1, (rowGap, columnGap) -> rowGap == -2 && columnGap == 1),
+	SSW(-2, -1, (rowGap, columnGap) -> rowGap == -2 && columnGap == -1),
+	NEE(1, 2, (rowGap, columnGap) -> rowGap == 1 && columnGap == 2),
+	NWW(1, -2, (rowGap, columnGap) -> rowGap == 1 && columnGap == -2),
+	SEE(-1, 2, (rowGap, columnGap) -> rowGap == -1 && columnGap == 2),
+	SWW(-1, -2, (rowGap, columnGap) -> rowGap == -1 && columnGap == -2);
+
+	private int rowGap;
+	private int columnGap;
+	private BiFunction<Integer, Integer, Boolean> find;
+
+	Direction(int rowGap, int columnGap, BiFunction<Integer, Integer, Boolean> find) {
+		this.rowGap = rowGap;
+		this.columnGap = columnGap;
+		this.find = find;
+	}
+
+	private static boolean isSameAbs(int rowGap, int columnGap) {
+		return Math.abs(rowGap) == Math.abs(columnGap);
+	}
+
+	public static Direction findDirection(Position position, Position targetPosition) {
+		int rowGap = position.calculateRowGap(targetPosition);
+		int columnGap = position.calculateColumnGap(targetPosition);
+
+		return Arrays.stream(values())
+			.filter(direction -> direction.find.apply(rowGap, columnGap))
+			.findFirst()
+			.orElseThrow(() -> new InvalidPositionException(InvalidPositionException.INVALID_DIRECTION));
+
+	}
+
+	public boolean hasPieceInRoute(Position position, Position targetPosition, Board board) {
+		List<Position> route = findRoutes(position, targetPosition);
+		return route.stream()
+			.map(board::findPiece)
+			.anyMatch(Optional::isPresent);
+	}
+
+	private List<Position> findRoutes(Position position, Position targetPosition) {
+		int loopCount = countOfLoop(position, targetPosition) - 1;
+		int routeRow = position.getRowNumber();
+		int routeColumn = position.getColumnNumber();
+		List<Position> route = new ArrayList<>();
+		for (int i = 0; i < loopCount; i++) {
+			routeRow += this.rowGap;
+			routeColumn += this.columnGap;
+			route.add(Position.of(routeColumn, routeRow));
+		}
+		return route;
+	}
+
+	private int countOfLoop(Position position, Position targetPosition) {
+		int columnGap = Math.abs(position.calculateColumnGap(targetPosition));
+		int rowGap = Math.abs(position.calculateRowGap(targetPosition));
+		return Math.max(columnGap, rowGap);
+	}
+
+	public static boolean isEveryDirection(Direction direction) {
```

Comment:
저 역시도 메서드 순서에 관해서 많이 생각을 해왔습니다..! 
처음에는 public - protected - private 순으로 메서드를 정렬하였는데, 이렇게 두니 가독성이 너무 떨어지더라구요! _(public에서 쓰이는 private 함수를 찾기 위해 저 아래까지 내려야가하는..)_ 
그래서 실제로 여러 컨벤션을 찾아보니, 접근지정자에 의한 순서보다는 논리적인 순서로 정렬하라고 나와있더라구요 :) 아래 링크들 참고해보시면 도움이 될 것 같아요!😊

 [Google Java Style Guide](https://google.github.io/styleguide/javaguide.html#s3.4-class-declaration)
> What is important is that each class uses some logical order, which its maintainer could explain if asked. For example, new methods are not just habitually added to the end of the class, as that would yield "chronological by date added" ordering, which is not a logical ordering.

[oracle컨벤션](https://www.oracle.com/java/technologies/javase/codeconventions-fileorganization.html#1852)


> These methods should be grouped by functionality rather than by scope or accessibility. For example, a private class method can be in between two public instance methods. The goal is to make reading and understanding the code easier.